spec/spec_helper.rb

```diff
@@ -10,7 +10,7 @@ def reset_config
 def load_config(whitelist_local = false)
   SafeRedirect.configure do |config|
     config.default_path = '/sdsdkkk'
-    config.domain_whitelists = %w{www.twitter.com www.bukalapak.com *.foo.org}
+    config.domain_whitelists = %w{http://www.twitter.com https://www.bukalapak.com http://*.foo.org https://*.test.com/good-end-point}
```

Comment:
Hi @sdsdkkk,

I forgot about another use case that was not supported in the current implementation.

For a whitelisted domain like `*.zoom.com/specific-domain`, we want `safe_domains?` to mark only the `*.zoom.com/specific-domain` but not `*.zoom.com`, or `*.zoom.com/other-domain` to be safe. Currently it is not the case.

Wondering what are your thoughts on this? I think this was also why I was parsing the entries in `domain_whitelists` as URI 😅 .

Reply:
I think that's a bit out of the scope of this gem since our only concern is ensuring that the redirection flow will not be tampered with so the user ended up being redirected to a malicious website by limiting the domain and sanitizing the redirection path string. While it's something that's nice to have, I haven't really come up with an idea on how to manage the path whitelists for the domain.

The changes in [this PR](https://github.com/sdsdkkk/safe_redirect/pull/14) should allow for more flexibility to the users of `safe_redirect` to extend the functionalities according to their own custom use cases e.g. by creating their own custom modules leveraging `safe_redirect`'s built-in functions, which should allow them to implement the path whitelisting as a custom extension of the `SafeRedirect` module.

But feel free to propose it if you have an idea that can handle this case without compromising the current configuration standards. If you haven't come up with it also, I think you can open an issue requesting this feature as a task tracker and we can try to think of a good approach to implement it for the future build.